### PR TITLE
DSND-2626: Composite resolution deletion

### PR DESCRIPTION
**Pull request by cgregoryCH**
## Describe the changes
* Allows deletion of singular resolutions within a composite.
* Document is deleted if the last resolution of a composite is removed.
* Feature flag included to disable this behaviour if set e.g. ```DELETE_CHILD_TRANSACTIONS_DISABLED=true```
* Mongo query is an aggregation which utilises Mongo indexes for child array entity_id's to retrieve the whole document along with child object indexes.

### Related Jira tickets

[DSND-2367](https://companieshouse.atlassian.net/browse/DSND-2367)

### General

- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing

- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation

_Where possible, add links to the respective Jira tickets when an item is checked_

- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to deployment repositories?

## Notes to the tester

_Add any testing hints or instructions here._

[DSND-2367]: https://companieshouse.atlassian.net/browse/DSND-2367?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by sthompsonCH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/delete/DeleteMapperDelegatorAspect.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.filinghistory.api.FilingHistoryApplication;
+import uk.gov.companieshouse.filinghistory.api.logging.DataMapHolder;
+import uk.gov.companieshouse.filinghistory.api.model.mongo.FilingHistoryDocument;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Aspect
+@Component
+@ConditionalOnProperty(prefix = "feature", name = "delete_child_transactions.disabled")
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/delete/DeleteMapperDelegatorAspect.java:16]: Would it be safer to be specific with the value?  e.g.
```
@ConditionalOnProperty(prefix = "feature", name = "delete_child_transactions.disabled", havingValue = "true")
```

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by johncook-ch — APPROVED** [review]

**Review by johncook-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**